tests/model/ExternalReferenceList.test.js

```diff
@@ -0,0 +1,53 @@
+/* eslint-env jest */
+
+/*
+ * This file is part of CycloneDX Node Module.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) OWASP Foundation. All Rights Reserved.
+ */
+
+const ExternalReferenceList = require('../../model/ExternalReferenceList')
+
+test('Homepage should not be present in externalReferences array if solely a period', () => {
+  const includedHomepageUrls = [
+    'http://someurl.com',
+    'https://someurl.com',
+    'https://any.other.string',
+    'https://example.com/#something.',
+    'https://example.com/?foobar=.',
+    'https://example.com/foo/.',
+    'https://example.com/bar.'
+  ]
+
+  const omittedHomepageUrls = [
+    'http://.',
+    'https://.'
+  ]
+
+  includedHomepageUrls.forEach(homepageUrl => {
```

Comment:
do not iterate over the list, but make each test an individual.
Reasons: if one test case in the list failed, the ones that come after that failed input will not be tested, but skipped.

therefore, restructure the test to a setup with `describe("%s is a homepage").each()`. 
see https://jestjs.io/docs/api#describeeachtablename-fn-timeout

Reply:
reminder: you can nest `describe()` to group tests. like 
```
describe('Homepage should not be present in externalReferences array if solely a period', () => {
  describe("test that homepage is in list", { 
  // ...  
 })
  describe("test that homepage with '.' is omitted", { 
  // ...  
 })
})
```